test/framework/src/main/java/org/elasticsearch/bootstrap/BootstrapForTesting.java

```diff
@@ -161,6 +166,18 @@ public boolean implies(ProtectionDomain domain, Permission permission) {
         }
     }
 
+    /** Add the codebase url of the given classname to the codebases map, if the class exists. */
+    private static void addClassCodebase(Map<String, URL> codebases, String name, String classname) {
+        try {
+            Class clazz = BootstrapForTesting.class.getClassLoader().loadClass(classname);
+            if (codebases.put(name, clazz.getProtectionDomain().getCodeSource().getLocation()) != null) {
+                throw new IllegalStateException("Already added " + name + " codebase for testing");
+            }
+        } catch (ClassNotFoundException e) {
+            // no class, fall through to not add
```

Comment:
When can this happen that it's okay to continue?

Reply:
It would happen in any tests other than core, since we only add the plugin-classloader jar to the core tests runtime classpath. I'll add a comment.